homeassistant/components/sensor/meteo_france.py

```diff
@@ -0,0 +1,140 @@
+"""
+Support for Meteo France raining forecast.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.meteo_france/
+"""
+
+import logging
+import datetime
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, ATTR_ATTRIBUTION, TEMP_CELSIUS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['meteofrance==0.2.2']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by Meteo-France"
+CONF_POSTAL_CODE = 'postal_code'
+
+STATE_ATTR_FORECAST = 'Forecast'
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+SENSOR_TYPES = {
+    'rain_chance': ['Rain chance', '%'],
+    'freeze_chance': ['Freeze chance', '%'],
+    'thunder_chance': ['Thunder chance', '%'],
+    'snow_chance': ['Snow chance', '%'],
+    'weather': ['Weather', None],
+    'wind_speed': ['Wind Speed', 'km/h'],
+    'next_rain': ['Next rain', 'min'],
+    'temperature': ['Temperature', TEMP_CELSIUS],
+    'uv': ['UV', None],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_POSTAL_CODE): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Meteo-France sensor."""
+    postal_code = config[CONF_POSTAL_CODE]
+
+    from meteofrance.client import meteofranceClient, meteofranceError
+
+    try:
+        meteofrance_client = meteofranceClient(postal_code)
+    except meteofranceError as exp:
+        _LOGGER.error(exp)
+        return
+
+    client = MeteoFranceUpdater(meteofrance_client)
+
+    add_entities([MeteoFranceSensor('Météo-France', variable, client)
+                  for variable in config[CONF_MONITORED_CONDITIONS]])
+
+
+class MeteoFranceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, name, condition, client):
+        """Initialize the sensor."""
+        self._name = name
+        self._condition = condition
+        self._client = client
+        self._data = client.get_data()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "{} {}".format(self._data["name"],
+                              SENSOR_TYPES[self._condition][0])
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        try:
+            return self._data[self._condition]
+        except KeyError:
+            _LOGGER.error("No result for the condition `%s`", self._condition)
```

Comment:
Please don't log in entity properties. We don't want any side effects here.

Reply:
Thanks for the review.
Some cities doesn't have all the conditions returned by the package. What should I do about the sensor then ?
Without the exception catching, it fails, generates an error and the sensor is not displayed on the front-end.
I can also keep the exception catching without logging. Sensors stay on the front-end with a `None` value and it's up to the user to remove the monitored-condition that is not available for this location. But without error logging, there is not hint that it's the problem

Maybe another solution would be to still catch those KeyError and set the state as "no data available for this location".

Reply:
Just move the access an exception handling etc to `update` and save the state in an instance attribute. Return the instance attributes here in the `state` property.

Reply:
latest commit should fix this